weblate/utils/search.py

```diff
@@ -470,7 +470,7 @@ def has_field(self, text: str, context: dict) -> Q:  # noqa: C901
         if text == "translation":
             return Q(state__gte=STATE_TRANSLATED)
         if text in {"variant", "shaping"}:
-            return Q(variant__isnull=False)
+            return Q(defined_variants__isnull=False)
```

Comment:
AFAIR this covers only manually created variants, not the regex based on the key. Can you please add tests for both cases to check that?